ci-operator/step-registry/upi/deprovision/azure/upi-deprovision-azure-commands.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set +e
+echo "Deprovisioning cluster ..."
+if [[ -f ${SHARED_DIR}/metadata.json ]]; then
+    INFRA_ID="$(jq -r .infraID ${SHARED_DIR}/metadata.json)"
+    AZURE_AUTH_CLIENT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_ID)
+    AZURE_AUTH_CLIENT_SECRET=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_SECRET)
+    AZURE_AUTH_TENANT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_TENANT_ID)
+    AZURE_SUBSCRIPTION_ID=$(cat ${SHARED_DIR}/AZURE_SUBSCRIPTION_ID)
+    az login --service-principal -u $AZURE_AUTH_CLIENT_ID -p "$AZURE_AUTH_CLIENT_SECRET" --tenant $AZURE_AUTH_TENANT_ID --output none
+    az account set --subscription ${AZURE_SUBSCRIPTION_ID}
+
+    
+    if az group list --query "[?name==${INFRA_ID}-rg]" != "[]" ; then
```

Comment:
You need single quotes around the resource group name.
```suggestion
    if az group list --query "[?name=='${INFRA_ID}-rg']" != "[]" ; then
```

You'll want to wrap the left-hand side in quotes. I also suggest using jq to make it a bit more clear what we are checking (or using wc if you want to use tsv output).
```suggestion
    if [ "$(az group list --query "[?name=='${INFRA_ID}-rg']" | jq length)" -ne 0 ]  ; then
```